compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/Utils.desktop.kt

```diff
@@ -71,22 +72,39 @@ internal fun getTransparentWindowBackground(
     return if (isWindowTransparent && !skikoTransparentWindowHack) java.awt.Color(0, 0, 0, 0) else null
 }
 
-internal fun JComponent.setTransparent(transparent: Boolean) {
-    /*
-     * Windows makes clicks on transparent pixels fall through, but it doesn't work
-     * with GPU accelerated rendering since this check requires having access to pixels from CPU.
-     *
-     * JVM doesn't allow override this behaviour with low-level windows methods, so hack this in this way.
-     * Based on tests, it doesn't affect resulting pixel color.
-     *
-     * Note: Do not set isOpaque = false for this container
-     */
-    if (transparent && hostOs == OS.Windows) {
-        background = java.awt.Color(0, 0, 0, 1)
-        isOpaque = true
-    } else {
-        background = null
-        isOpaque = false
+
+/**
+ * Windows makes clicks on transparent pixels fall through, but it doesn't work
+ * with GPU accelerated rendering since this check requires having access to pixels from CPU.
+ *
+ * JVM doesn't allow override this behaviour with low-level windows methods, so hack this by filling
+ * the background with an almost transparent color.
+ * Based on tests, it doesn't affect resulting pixel color.
+ */
+internal open class JLayeredPaneWithTransparencyHack: JLayeredPane() {
```

Comment:
Let's move it into a separate file

Reply:
Why? It's a tiny, internal utility.

Reply:
Unlike general "utils" and 1-liner functions, this one is already separated, named entity.
But it's not a blocker of course.

Reply:
I don't think just being a class instead of a function warrants moving it to its own file.